### PR TITLE
[Unit-tests] wrote unit-tests for profile-service / task#2604

### PR DESCRIPTION
**Pull request by yuriiLisunov**
**Description:**
related to:
ita-social-projects/GreenCity#2604
Part 3 (src/app/component/user/components/profile/profile-service/profile-service.ts).

### CODE REVIEW

**Review by idzhersh — APPROVED** [review]

**Review by VitalyBorysenko — APPROVED** [review]

File: src/app/component/user/components/profile/profile-service/profile.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        expect(info.content).toBe('Great day!');
+      });
+
+      const req = httpMock.expectOne(`https://greencity.azurewebsites.net/factoftheday/?lang=en`);
+      expect(req.request.method).toBe('GET');
+      req.flush(fact);
+    });
+  });
+
+  describe('getUserInfo', () => {
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-service/profile.service.spec.ts:66]: getUserInfo - please provide an appropriate name to describe test case

File: src/app/component/user/components/profile/profile-service/profile.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        showLocation: false,
+        showShoppingList: true,
+        socialNetworks: []
+      };
+
+      profileService.getUserInfo().subscribe(info => {
+        expect(info.rating).toBe(1999);
+      });
+
+      const req = httpMock.expectOne(`https://greencity-user.azurewebsites.net/user/1111/profile/`);
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-service/profile.service.spec.ts:84]: https://greencity-user.azurewebsites.net/user/ -> make constant with this at line 11

**Review by paslavskyi9roman — APPROVED** [review]

File: src/app/component/user/components/profile/profile-service/profile.service.ts [hunk 3]
```diff
@@ -40,14 +40,15 @@ export class ProfileService {
...
-    this.setUserId();
     const currentLang = this.languageService.getCurrentLanguage();
+    this.setUserId();
     return this.http.get<ShoppingList[]>(`
     ${mainLink}custom/shopping-list-items/${this.userId}/custom-shopping-list-items?lang=${currentLang}
     `);
   }
 
   public getUserProfileStatistics(): Observable<ProfileStatistics> {
+    this.setUserId();
```
Comment (JimHolman1199) [inline src/app/component/user/components/profile/profile-service/profile.service.ts:51]: why do we need to call this method here ?

Comment (yuriiLisunov) [reply src/app/component/user/components/profile/profile-service/profile.service.ts:51]: because we get statistics of profile by user id

**Review by KirillAbramo — APPROVED** [review]

**Review by AlekOK — APPROVED** [review]

**Review by mr-pavlenko — APPROVED** [review]

**Pull request merged**